rtl/eclrtl/rtlformat.cpp

```diff
@@ -766,7 +766,9 @@ void CommonJsonWriter::outputBeginNested(const char *fieldname, bool nestChildre
         appendJSONName(out, fieldname);
     }
     out.append("{");
-    if (!nestChildren && !nestLimit)
+    if ((flags & XWFnonestchildren) && !(flags & XWFnoindent))
+        nestLimit++;
```

Comment:
I haven't really understood the original meaning of nestLimit - that might be because scanning through what they did (before this PR) was not what I expected.

The variable declarations could do with some comments on them, briefly describing their semantics.

e.g. it's not clear to why when outputEndNested hit indent==nestLimit, it reset nestLimit = 0 (which means off?).
Should it support arbitrary depth nesting?

Reply:
I think as you've eluded to in comments in meeting, current behaviour is buggy, returning nestIndent to 0 after inner most indentation is close (?)

With this change and with XWFnonestchildren on, instead of setting nestLimit to current indent level, it will increment and decrement it - which will prevent any nesting of child rows/datasets ? because won't return to 0 until out of it.

(also not sure what effect CommonJsonWriter::cutFrom has, when it sets nestLimit)

Reply:
nestLimit is really badly named.  The original nestLimit was an attempt at emulating what the XmlWriter did, but not very well.  To fix the feature it I think we need more state keeping track of the stack of how outputBeginNested()  and outputEndNested() are called throughout the json structure.

A future change could fix outputBeginNested(..., nestChildren=true); Yes, "prevent any nesting of child rows/datasets". .. and prevent the effect that happens to nested records even if nestChildren wasn't set.  The current change is meant to be a very low risk way of working in a mode that will support despray.  I think despray can only handle specific placement of newlines.  This will fix ECL JSON file output in a way that is despray compatible.

I'm going to change the name of the flag to be something that more reflects the desired outcome, rather than reflecting "turn off nestChildren".